### PR TITLE
[CN DateTimeV2 JS] Add support for relative cases like "5天前"

### LINKED ISSUE #1885: [CN DateTimeV2 Python] ”五天前“ , "5天前” and "2小时前" are not extracted
- Platform: Python
 - Version of package: recognizers-text-date-time 1.0.1a0

```python
>>> from recognizers_date_time import DateTimeRecognizer, Culture

>>> model = DateTimeRecognizer(Culture.Chinese).get_datetime_model()

>>> model.parse('5天前')
[]
>>> model.parse('五天前')
[]
>>> model.parse('2小时前')
[]
```

### PR DESCRIPTION
**Pull request by Grey0202**
Fixes #1885 in JavaScript. Add support case for "5天前", ”5小时后“ in JS.

### CODE REVIEW

File: JavaScript/packages/recognizers-date-time/src/dateTime/chinese/dateConfiguration.ts [hunk 1]
```diff
@@ -227,7 +241,7 @@ export class ChineseDateParser extends BaseDateParser {
             }
             if (!innerResult.success) {
                 // TODO create test
-                innerResult = this.parserDurationWithAgoAndLater(source, referenceDate);
+                innerResult = this.parserDurationWithBeforeAndAfter(source, referenceDate);
```
Comment (tellarin) [inline JavaScript/packages/recognizers-date-time/src/dateTime/chinese/dateConfiguration.ts]: Is this really correct? We don't need both?

File: JavaScript/packages/recognizers-date-time/src/dateTime/chinese/dateConfiguration.ts [hunk 2]
```diff
@@ -512,6 +526,16 @@ export class ChineseDateParser extends BaseDateParser {
         return result;
     }
 
+    private convertChineseToNumber(source: string): number {
```
Comment (tellarin) [inline JavaScript/packages/recognizers-date-time/src/dateTime/chinese/dateConfiguration.ts]: Chinese what to number?

File: JavaScript/packages/recognizers-date-time/src/dateTime/chinese/dateTimeConfiguration.ts [hunk 3]
```diff
@@ -349,4 +389,84 @@ export class ChineseDateTimeParser extends BaseDateTimeParser {
 
         return ret;
     }
+
+    // Handle cases like "5分钟前", "1小时以后"
+    private parserDurationWithBeforeAndAfter(source: string, referenceDate: Date): DateTimeResolutionResult {
```
Comment (tellarin) [inline JavaScript/packages/recognizers-date-time/src/dateTime/chinese/dateTimeConfiguration.ts]: Is this just the same repetition of this method?
We should look into unifying implementations. Maybe as an util like in .NET?

File: JavaScript/packages/recognizers-date-time/src/dateTime/chinese/dateConfiguration.ts (hunk 1 above)
Comment (tellarin) [reply JavaScript/packages/recognizers-date-time/src/dateTime/chinese/dateConfiguration.ts]: Or is this more like a rename and the other method is no longer needed?

Comment (Grey0202) [reply JavaScript/packages/recognizers-date-time/src/dateTime/chinese/dateConfiguration.ts]: "parserDurationWithAgoAndLater" is used in English and parserDurationWithBeforeAndAfter is for Chinese.

Comment (tellarin) [reply JavaScript/packages/recognizers-date-time/src/dateTime/chinese/dateConfiguration.ts]: Why not unify the method names?

**Comment by gengchen528** [comment]
这个问题解决了么，我现在把版本改成了1.2.7，发现还是无法解析‘5天后’‘5分钟前’这样的词语，返回的数据为空，`js`版本，`node>10`

**Comment by tellarin** [comment]
@gengchen528, this PR is still pending fixes and has not been merged. Once it's merged, it will be available in the next version of the NPM packages.

**Comment by gengchen528** [comment]
> @gengchen528, this PR is still pending fixes and has not been merged. Once it's merged, it will be available in the next version of the NPM packages.

ok   thank you

**Pull request merged**